cern_access/indico_cern_access/forms.py

```diff
@@ -105,6 +106,16 @@ class AccessIdentityDataForm(IndicoForm):
     birth_date = IndicoDateField(_('Birth date'), [DataRequired()])
     nationality = SelectField(_('Country of birth'), [DataRequired()])
     birth_place = StringField(_('Place of birth'), [DataRequired()])
+    by_car = BooleanField(_('Are you bringing your own car?'), [Optional()], widget=SwitchWidget(_('Yes'), _('No')),)
+    license_plate = StringField(
+        _('License plate'),
+        [
+            HiddenUnless('by_car'),
+            Length(min=3),
+            IndicoRegexp(r'^[0-9A-Za-z]+([- ][ ]*[0-9A-Za-z]+)*$',
```

Comment:
I tried changing this to `IndicoRegexp` which adds the `pattern` attribute to the `<input>`  element for client-side validation, but the form is submitted anyway. Probably because submit is going through loads of JS in the legacy angular mess...

In any case, this means the regform data is lost if validation on this field fails, which isn't amazing. Maybe we can find some way to abort submission of the registration form if validation fails here?

Reply:
What do you mean? It doesn't seem like it's lost:

Reply:
I mean the data in the actual regform, not related to site access

Reply:
Ah right.

Reply:
OK, what I've just pushed should address this. I'm not particularly proud of it, but anyway, it's CERN-specific and it's working with a legacy module. It does a quick check client-side, to avoid reloading the form. It doesn't produce error messages, however. It only sets the fields in red.